src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1512,3 +1512,42 @@ def list_consumption_locations():
     client = web_client_factory()
     regions = client.list_geo_regions(sku='Dynamic')
     return [{'name': x.name.lower().replace(" ", "")} for x in regions]
+
+
+def enable_zip_deploy(resource_group_name, name, src, slot=None):
+    # if src is not of .zip extension throw exception
+    if not src.endswith('.zip'):
+        raise CLIError("src '{}' path needs to be .zip file".format(src))
```

Comment:
This should be done as a validator instead of here.
You should also check that the file path actually exists.

Also, should use os.path.realpath(os.path.expanduser(src)) to allow users to specify source as `~/myzip.zip`

Reply:
Validators can raise errors if the argument is correct.